internal/testcase/ecs_pod_creator.go

```diff
@@ -0,0 +1,185 @@
+package testcase
+
+import (
+	"context"
+	"testing"
+
+	"github.com/evergreen-ci/cocoa"
+	"github.com/evergreen-ci/cocoa/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// ECSPodCreatorTestCase represents a test case for a cocoa.ECSPodCreator.
+type ECSPodCreatorTestCase func(ctx context.Context, t *testing.T, c cocoa.ECSPodCreator)
+
+// ECSPodCreatorNoVaultTests returns common test cases that a cocoa.ECSPodCreator should support.
+func ECSPodCreatorNoVaultTests() map[string]ECSPodCreatorTestCase {
+	return map[string]ECSPodCreatorTestCase{
+		"CreatePodFailsWithInvalidCreationOpts": func(ctx context.Context, t *testing.T, c cocoa.ECSPodCreator) {
+			opts := cocoa.NewECSPodCreationOptions()
+
+			p, err := c.CreatePod(ctx, opts)
+			require.Error(t, err)
+			require.Zero(t, p)
+		},
+		"CreatePodFailsWithSecretsNoVault": func(ctx context.Context, t *testing.T, c cocoa.ECSPodCreator) {
+			envVar := cocoa.NewEnvironmentVariable().
+				SetName("envVar").
+				SetSecretOptions(*cocoa.NewSecretOptions().
+					SetName(testutil.NewSecretName("name")).
+					SetValue("value"))
+			containerDef := cocoa.NewECSContainerDefinition().
+				SetImage("image").
+				SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+				SetName("container")
+			require.NotNil(t, containerDef.EnvVars)
+
+			execOpts := cocoa.NewECSPodExecutionOptions().SetCluster(testutil.ECSClusterName()).SetExecutionRole(testutil.ExecutionRole())
+			assert.NoError(t, execOpts.Validate())
+
+			opts := cocoa.NewECSPodCreationOptions().
+				SetName(testutil.NewTaskDefinitionFamily("name")).
```

Comment:
Can you use `t.Name()` in the task definition to make it easier to debug when task definitions get leaked? The tests are currently leaking a lot of ECS task definitions and leaving them behind in ECS.